File: pulsar-tools/pulsar-browser/src/main/java/com/github/kklisura/cdt/protocol/v2023/types/webaudio/AudioParam.java
package com.github.kklisura.cdt.protocol.v2023.types.webaudio;

/*-
 * #%L
 * cdt-java-client
 * %%
 * Copyright (C) 2018 - 2023 Kenan Klisura
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */

/** Protocol object for AudioParam */
public class AudioParam {

  private String paramId;

  private String nodeId;

  private String contextId;

  private String paramType;

  private AutomationRate rate;

  private Double defaultValue;

  private Double minValue;

  private Double maxValue;

  public String getParamId() {
    return paramId;
  }

  public void setParamId(String paramId) {
    this.paramId = paramId;
  }

  public String getNodeId() {
    return nodeId;
  }

  public void setNodeId(String nodeId) {
    this.nodeId = nodeId;
  }

  public String getContextId() {
    return contextId;
  }

  public void setContextId(String contextId) {
    this.contextId = contextId;
  }

  public String getParamType() {
    return paramType;
  }

  public void setParamType(String paramType) {
    this.paramType = paramType;
  }

  public AutomationRate getRate() {
    return rate;
  }

  public void setRate(AutomationRate rate) {
    this.rate = rate;
  }

  public Double getDefaultValue() {
    return defaultValue;
  }

  public void setDefaultValue(Double defaultValue) {
    this.defaultValue = defaultValue;
  }

  public Double getMinValue() {
    return minValue;
  }

  public void setMinValue(Double minValue) {
    this.minValue = minValue;
  }

  public Double getMaxValue() {
    return maxValue;
  }

  public void setMaxValue(Double maxValue) {
    this.maxValue = maxValue;
  }
}

File: pulsar-core/pulsar-tools/pulsar-browser/src/main/kotlin/ai/platon/cdt/kt/protocol/types/webaudio/AudioParam.kt
@file:Suppress("unused")
package ai.platon.cdt.kt.protocol.types.webaudio

import com.fasterxml.jackson.`annotation`.JsonProperty
import kotlin.Double
import kotlin.String

/**
 * Protocol object for AudioParam
 */
data class AudioParam(
  @param:JsonProperty("paramId")
  val paramId: String,
  @param:JsonProperty("nodeId")
  val nodeId: String,
  @param:JsonProperty("contextId")
  val contextId: String,
  @param:JsonProperty("paramType")
  val paramType: String,
  @param:JsonProperty("rate")
  val rate: AutomationRate,
  @param:JsonProperty("defaultValue")
  val defaultValue: Double,
  @param:JsonProperty("minValue")
  val minValue: Double,
  @param:JsonProperty("maxValue")
  val maxValue: Double,
)
